Source/Scene/Model.js

```diff
@@ -2972,11 +3069,58 @@ define([
                     castShadows : castShadows,
                     receiveShadows : receiveShadows,
                     uniformMap : uniformMap,
-                    renderState : rs,
+                    renderState : drawRS,
                     owner : owner,
                     pass : isTranslucent ? Pass.TRANSLUCENT : Pass.OPAQUE
                 });
 
+                // Setup the stencil command for the highlight
+                var highlightRS = clone(rs);
+                highlightRS.stencilTest = {
+                    enabled : true,
+                    frontFunction : WebGLConstants.NOTEQUAL,
+                    backFunction : WebGLConstants.NOTEQUAL,
+                    reference : 1,
+                    mask : ~0,
+                    frontOperation : {
+                        fail : WebGLConstants.KEEP,
+                        zFail : WebGLConstants.KEEP,
+                        zPass : WebGLConstants.REPLACE
+                    },
+                    backOperation : {
+                        fail : WebGLConstants.KEEP,
+                        zFail : WebGLConstants.KEEP,
+                        zPass : WebGLConstants.REPLACE
+                    }
+                };
+                highlightRS.cull = {
+                    enabled : false,
+                    face : WebGLConstants.FRONT
+                };
+                highlightRS.depthTest = false;
+
+                highlightRS = RenderState.fromCache(highlightRS);
+
+                // Setup the highlight color and size uniforms.
+                uniformMap.u_highlightColor = createHighlightColorFunction(model);
+                uniformMap.u_highlightSize = createHighlightSizeFunction(model);
+
+                var highlightCommand = new DrawCommand({
```

Comment:
I tried to mimic what was done with the pickCommands, so kept the highlight command and resource creation close to where the pick commands are generated.  If you don't mind I'll defer this to you to move around so I don't break anything :)